pkg/vars/vars_loader_test.go

```diff
@@ -645,7 +645,7 @@ func TestVarsLoader_AwsSecretsManager(t *testing.T) {
 
 	testVarsLoader(t, func(vl *VarsLoader, vc *VarsCtx, aws *aws.FakeAwsClientFactory) {
 		aws.Secrets = map[string]string{
-			"secret": `{"test1": {"test2": 42}}`,
+			"secret:secret": `{"test1": {"test2": 42}}`,
```

Comment:
Can you explain why this was needed?

Reply:
It was needed because the official ARN parser doesn't differentiate between resource type and resource id and puts everything in a single resource field. The fake_clientfactory uses the field `Resource` to lookup the secrets which was `secret` with the custom parser but is now `secret:secret`. The change has no effect on the productive code because you only read the region in case if its not passed to `GetAwsSecretsManagerSecret` at that stays the same. Eventually, you just pass the ARN string to `GetSecretValue`, so there is no change needed at that point.

Hope that helps :smile_cat: 

PS: As I wrote that down, I realized that we should maybe check at the beginning of `GetAwsSecretsManagerSecret` if its a valid ARN or not and not only when there is no region or much later when we try to get the secret. I will add that.

Reply:
I need to change a lot more. I will explain it to you tomorrow. :smile_cat: